### ISSUE TITLE
Cannot mask `Name` after `Name` in Python

### ISSUE THREAD

**Issue by mindol** [uiuc-focal-lab/syncode#143]
## Problem
Hello, I think `Name` is not properly masked right after another `Name` in Python.
Currently, I'm using `microsoft/phi-2` model and tokenizer.

## Example

1. At first, we start with the initial prompt:
```python
def f(x):
    If
```
Note that `If` is a Name; not a keyword because it starts with the uppercase letter.

2. Next, phi2 with Syncode generates a token ` x`, so the sequence becomes:
```python
def f(x):
    If x
```

3. One more time, phi2 with Syncode generates a token ` is`, so the sequence becomes:
```python
def f(x):
    If x is
```
This partial code is syntactically **not acceptable**. If I try to generate one more token, following error occurs:

```bash
Parsing failed! Falling back to unconstrained decoding.
Exception: Unexpected token Token('NAME', 'x') at line 2, column 8.
Expected one of:
        * COMMA
        * AT
        * IF
        * LPAR
        * __ANON_6
        * __ANON_17
        * ELSE
        * __ANON_13
        * __ANON_19
        * __ANON_8
        * CIRCUMFLEX
        * __ANON_16
        * __ANON_14
        * ASYNC
        * RPAR
        * PLUS
        * _NL
        * FROM
        * __ANON_2
        * __ANON_5
        * AND
        * __ANON_12
        * __ANON_20
        * AMPERSAND
        * __ANON_9
        * SLASH
        * IS
        * LESSTHAN
        * LSQB
        * __ANON_11
        * MORETHAN
        * EQUAL
        * PERCENT
        * STAR
        * FOR
        * __ANON_15
        * RBRACE
        * VBAR
        * __ANON_1
        * IN
        * __ANON_4
        * __ANON_7
        * __ANON_22
        * __ANON_21
        * __ANON_18
        * DOT
        * COLON
        * MINUS
        * OR
        * __ANON_10
        * RSQB
        * NOT
        * SEMICOLON
        * AS
        * __ANON_3

Partial code: def f(x):
    If x is
Parsed lexical tokens: [Token('DEF', 'def'), Token('NAME', 'f'), Token('LPAR', '('), Token('NAME', 'x'), Token('RPAR', ')'), Token('COLON', ':'), Token('_NL', '\n    '), Token('NAME', 'If'), Token('NAME', 'x')]
```

## Reproduction code
**Dependencies**:
syncode v0.1.4
torch==2.5.1
transformers==4.44.0

**Code**:
```python
from syncode.grammar_decoder import SyncodeLogitsProcessor
from syncode.parsers.grammars import Grammar
import torch
from transformers import AutoModelForCausalLM
from transformers import AutoTokenizer

# Load model and tokenizer
torch.set_default_device("cpu")

model_name = "microsoft/phi-2"
model = AutoModelForCausalLM.from_pretrained(model_name)
tokenizer = AutoTokenizer.from_pretrained(model_name)


def generate_one_token(prompt: str) -> str:
    # 1. Initialize SyncodeLogitsProcessor
    syncode_processor = SyncodeLogitsProcessor(grammar=Grammar("python"),
                                               tokenizer=tokenizer,
                                               parse_output_only=False)
    syncode_processor.reset(prompt)

    # 2. Generate scores with SyncodeLogitsProcessor
    inputs = tokenizer(prompt, return_tensors="pt").input_ids
    outputs = model.generate(inputs,
                             attention_mask=torch.ones_like(inputs),
                             logits_processor=[syncode_processor],
                             return_dict_in_generate=True,
                             output_scores=True,
                             max_new_tokens=1)

    # 3. Print scores
    scores = outputs.scores
    generated_id = outputs.sequences[0, -1]
    generated_str = tokenizer.decode(generated_id, skip_special_tokens=True)
    print(f"Generated token id: {generated_id} ({generated_str})")
    print(f"Score of this token: {scores[0][-1][generated_id]}")

    return generated_str


prompt = """def f(x):
    If"""
assert (gen1 := generate_one_token(prompt)) == " x"
assert (gen2 := generate_one_token(prompt + gen1)) == " is"
generate_one_token(prompt + gen1 + gen2)
```

**Comment by shubhamugare** [comment]
Hi @mindol,

Thanks for filing the issue. It should be fixed now in the recent PR.
